src/messages.ts

```diff
@@ -543,20 +549,73 @@ export class Messages<T extends string> {
+   * @param actionTokens The action messages tokens.
+   */
+  public createWarning(key: T, tokens: Tokens = [], actionTokens: Tokens = []): StructuredMessage {
+    const [message, name, actions] = this.formatMessageContents('warning', key, tokens, actionTokens);
+    return { message, name, actions };
+  }
+
+  /**
+   * Convenience method to create info using message labels.
+   *
+   * `info.name` will be the upper-cased key, remove prefixed `info.` and will always end in Info.
+   * `info.actions` will be loaded using `${key}.actions` if available.
+   *
+   * @param key The key of the warning message.
+   * @param tokens The warning message tokens.
+   * @param actionTokens The action messages tokens.
+   */
+  public createInfo(key: T, tokens: Tokens = [], actionTokens: Tokens = []): StructuredMessage {
+    const [message, name, actions] = this.formatMessageContents('info', key, tokens, actionTokens);
+    return { message, name, actions };
+  }
+
+  /**
+   * Formats message contents given a message type, key, tokens and actions tokens
+   *
+   * `<type>.name` will be the upper-cased key, remove prefixed `<type>.` and will always end in 'Error | Warning | Info.
+   * `<type>.actions` will be loaded using `${key}.actions` if available.
+   *
+   * @param type The type of the message set must 'error' | 'warning' | 'info'.
+   * @param key The key of the warning message.
+   * @param tokens The warning message tokens.
+   * @param actionTokens The action messages tokens.
+   */
+  private formatMessageContents(
+    type: 'error' | 'warning' | 'info',
+    key: T,
+    tokens: Tokens = [],
+    actionTokens: Tokens = []
+  ): [string, string, string[] | undefined] {
+    const label = upperFirst(type);
+    const labelRegExp = new RegExp(`${label}$`);
+    const searchValue: RegExp = type === 'error' ? /^error.*\./ : new RegExp(`^${type}.`);
     // Convert key to name:
-    //     'myMessage' -> `MyMessageError`
-    //     'myMessageError' -> `MyMessageError`
-    //     'error.myMessage' -> `MyMessageError`
-    //     'errors.myMessage' -> `MyMessageError`
-    const errName = `${upperFirst(key.replace(/^errors*\./, ''))}${/Error$/.exec(key) ? '' : 'Error'}`;
-    const errMessage = this.getMessage(key, tokens);
-    let errActions;
+    //     'myMessage' -> `MyMessageWarning`
+    //     'myMessageError' -> `MyMessageWarning`
+    //     'warning.myMessage' -> `MyMessageWarning`
+    const name = `${upperFirst(key.replace(searchValue, ''))}${labelRegExp.exec(key) ? '' : label}`;
+    const message = this.getMessage(key, tokens);
+    let actions;
     try {
-      errActions = this.getMessageWithMap(`${key}.actions`, actionTokens, this.messages);
+      actions = this.getMessageWithMap(`${key}.actions`, actionTokens, this.messages);
     } catch (e) {
       /* just ignore if actions aren't found */
     }
-    return new SfdxError(errMessage, errName, errActions, exitCodeOrCause, cause);
+    return [message, name, actions];
```

Comment:
I think it'd be more idiomatic to return an object instead, `{ message, name, actions }`

That way you can simplify the `createInfo` and `createWarning` methods:

```typescript
createInfo(key, ...) {
  return this.formatMessageContents(key, ...)
}
```